pkg/services/sqlstore/monitor.go

```diff
@@ -44,6 +45,20 @@ type MonitorWithCollectorDTO struct {
 	Created         time.Time
 }
 
+// scrutinizeState fixes the state.  We can't just trust what the database says, we have to verify that the value actually has been updated recently.
+// this could later be made more precise by taking into account the current scheduler offset, the range it is assumed to be in (say 0 to 50 seconds)
+// and the offset of the monitor.  But for now, we simply allow 1 run and an additional minute of delay.
+func scrutinizeState(state m.CheckEvalResult, stateCheck time.Time, frequency int64) m.CheckEvalResult {
+	if state == m.EvalResultUnknown {
+		return state
+	}
+	maxDelay := time.Duration(frequency+60) * time.Second
```

Comment:
I personally would prefer to have the maxDelay be a multiple of the frequency.  ie, the state should be unknown if 2 consecutive checks failed to run.